app/Jobs/GenerateSimilarityIndex.php

```diff
@@ -79,25 +79,32 @@ public function __construct(Volume $volume)
     public function handle()
     {
         $collection = $this->volume->images;
-        // check if all images has hash, empty is true if all has hash
-
-        if ($collection->whereNull('hash')->isEmpty()) {
-            $imageHashArray = $collection->pluck('hash', 'id');
-            $similarityIndexArray = $this->createSimilarityIndexArray($imageHashArray);
-            if (!is_null($similarityIndexArray)) {
-                foreach ($this->volume->images as $image) {
-                    $id = $image->id;
-                    $similarityIndex = $similarityIndexArray[$id];
-                    $image->similarityIndex = $similarityIndex;
-                    $image->save();
+        if ($collection->count() !== 1) {
+            if ($collection->whereNull('hash')->isEmpty()) {
+                $imageHashArray = $collection->pluck('hash', 'id');
+                $similarityIndexArray = $this->createSimilarityIndexArray($imageHashArray);
+                if (!is_null($similarityIndexArray)) {
+                    foreach ($this->volume->images as $image) {
+                        $id = $image->id;
+                        $similarityIndex = $similarityIndexArray[$id];
+                        $image->similarityIndex = $similarityIndex;
+                        $image->save();
+                    }
                 }
             }
+        } else {
+            foreach ($this->volume->images as $image) {
+                $image->similarityIndex = 0;
+                $image->save();
+            }
         }
```

Comment:
Just a tip I read somewhere some time ago. Special cases can be caught with `if` and `return` statements before the "default" action is taken. This makes code more readable (i.e. keeps indentation in check). Example:

```php
if ($collection->whereNull('hash')->isNotEmpty()) {
   // maybe create a warning in the logs
   return;
}

if ($collection->count() <= 1) {
   // do special stuff
   return;
}

// do default stuff
```